### PR TITLE
redirect stderr to a file for debugging

### PR DESCRIPTION
**Pull request by hideyukn88**
Redirect stderr to a file for debugging

### CODE REVIEW

**Review by benhillis — APPROVED** [review]

**Review by spronovo — APPROVED** [review]

File: WSLGd/main.cpp [hunk 1]
```diff
@@ -76,14 +77,11 @@ int main(int Argc, char *Argv[])
...
-    //
-    // N.B. fprintf must be used instead of dprintf because glibc does not correctly
-    //      handle /dev/kmsg.
+    // Open a file for logging errors and set it to stderr for WSLGd as well as any child process.
     {
-        wil::unique_fd logFd(TEMP_FAILURE_RETRY(open("/dev/kmsg", (O_WRONLY | O_CLOEXEC))));
-        if ((logFd) && (logFd.get() != STDERR_FILENO)) {
-            THROW_LAST_ERROR_IF(dup3(logFd.get(), STDERR_FILENO, O_CLOEXEC) < 0);
+        wil::unique_fd fd_stdErrLog(open(c_stdErrLogMount, (O_RDWR | O_CREAT), (S_IRUSR | S_IRGRP | S_IROTH)));
+        if (fd_stdErrLog && fd_stdErrLog.get()) {
```
Comment (benhillis) [inline WSLGd/main.cpp:83]: Why the separate check for fdStdErrLog.get()?  0 is a valid file descriptor.

Comment (benhillis) [reply WSLGd/main.cpp:83]: did you mean to check for != STDERR_FILENO?

**Review by benhillis — CHANGES_REQUESTED** [review]

**Comment by Biswa96** [comment]
Wouldn't it be better if there is an option either to log in a text file or in a console window? Like `.wslconfig` has `debugConsole` xor `debugConsoleLogFile`.

File: WSLGd/main.cpp (hunk 1 above)
Comment (hideyukn88) [reply WSLGd/main.cpp:83]: yes, I will change to != STDERR_FILENO.

**Comment by hideyukn88** [comment]
> Wouldn't it be better if there is an option either to log in a text file or in a console window? Like `.wslconfig` has `debugConsole` xor `debugConsoleLogFile`.

This module doesn't have any visibility to .wslconfig, so that's not possible. The immediate goal for us is to capture any error sent to stderr, during WSLg development. Once stable, it's possible we may switch back to kmsg, including child process (previously only this WSLGd process's stderr was redirected).

**Review by benhillis — APPROVED** [review]

**Pull request merged**